shared/js/background/classes/tracker-stats.js

```diff
@@ -1,192 +1,224 @@
+            const packIsYoungerThanADay = (now - pack.end) < DAY
+            if (!packIsYoungerThanADay) continue
+            for (const [key, count] of Object.entries(pack.entries)) {
+                output[key] = (output[key] ?? 0) + count
             }
-            if (count > 0) {
-                stats.push({ key, count })
+        }
+
+        // aggregate current, but only if it's younger than a day
+        const currentIsYoungerThanADay = (now - this.current.start) < DAY
+        if (currentIsYoungerThanADay) {
+            for (const [key, count] of this.current.entries) {
+                output[key] = (output[key] ?? 0) + count
             }
         }
 
-        // sort them, so that the highest count's are first
-        const sorted = stats.sort((a, b) => b.count - a.count)
+        return output
+    }
 
-        // now it's sorted, we need to partition the array based on the maxCount
-        const results = sorted.slice(0, maxCount)
+    /**
+     * @param {number} [now] optional timestamp to use as the current time
+     * @returns {SortedOutput}
+     */
+    sorted (now = Date.now()) {
+        const data = this.data(now)
 
-        // everything else will make up the 'overflow' count that can be used to
-        // display a category like 'Other'
-        const overflowItems = sorted.slice(maxCount)
-        const overflow = overflowItems.reduce((sum, item) => sum + item.count, 0)
+        /** @type {SortedOutput} */
+        const keys = Object.keys(data)
+            .map(key => {
+                return { key, count: data[key] }
+            })
+            .sort((a, b) => b.count - a.count)
 
-        return {
-            results,
-            overflow
-        }
+        return keys
     }
 
     /**
-     * Allow our internal data to be 'set'.
-     *
-     * We are very strict about the incoming data here, mostly treating
-     * it as untrusted.
-     *
-     * It should be in this format:
-     *
-     * {
-     *     "totalCount": 12,
-     *     "entries": {
-     *         "Google": [1673438914778],
-     *         "Facebook": [1673438914778, 1673438914778, 1673438914792]
-     *     }
-     * }
-     *
-     * This is normally produced by `.serialize()` on this instance (see below)
-     *
      * @param {object} params
      * @param {number} params.totalCount
-     * @param {Record<string, number[]>} params.entries
-     * @throws
+     * @param {Pack} params.current
+     * @param {Pack[]} params.packs
      */
     deserialize (params) {
-        // define what shape incoming data must be in
+        // this is the schema for a single pack. It's used for the current pack and for the packs array
+        const packSchema = z.object({
+            start: z.number().default(0),
+            end: z.number().default(0),
+            entries: z.record(z.string(), z.number())
+        })
+
+        // this is the schema for the entire storage object
         const storageSchema = z.object({
```

Comment:
Do we have to handle users with existing storage in the old format? Will this just get silently deleted and replaced with the new format?